pandas/tests/plotting/test_frame.py

```diff
@@ -1129,14 +1129,18 @@ def test_bar_categorical(self):
         for df in [df1, df2]:
             ax = df.plot.bar()
             ticks = ax.xaxis.get_ticklocs()
-            tm.assert_numpy_array_equal(ticks, np.array([0, 1, 2, 3, 4, 5]))
+            tm.assert_numpy_array_equal(
+                ticks, np.array([0, 1, 2, 3, 4, 5], dtype=np.float)
```

Comment:
Why did this need to change?

Reply:
The ticks pos are handled by matplotlib fonction convert_xunits, which return a float array rather than the int array previously used.
I would argue that having float array for tick position make more sense to have the same type for all plot. 
I do not think that the point of the test was to ensure the type of the tick position, so I changed it.

Reply:
Can you help me understand what part of this PR would have changed this? Still not sure on this

Reply:
Before the PR, the tick position where defined by using `self.tick_pos = np.arange(len(data))`.
In this PR, I let matplotlib handle the position by calling
```
ax.xaxis.update_units(self.ax_index)
self.tick_pos = ax.convert_xunits(self.ax_index)
```
This use the matplotlib Conversion Interface. For convertiong string category, this use numpy float array (see https://matplotlib.org/_modules/matplotlib/category.html#StrCategoryConverter).

I could cast this back to an int array, but it made more sense IMO to change the test here, that trying to force the tick position to be integer.

Reply:
I think this should stay as integer to not break backwards compatibility - how hard would it be to accomplish that?